src/rtpghi~.c

```diff
@@ -12,80 +12,65 @@
 static t_class *rtpghi_tilde_class;
 
 typedef struct _rtpghi_tilde {
-    t_object  x_obj;
-    t_float W_pd;
-    t_float ol_pd;
-    t_float M_pd;
-    t_float tol_pd;
-    t_float do_causal_pd;
-    t_symbol window_type;
-    phaseret_rtpghi_state_s* sta_pd;
-    ltfat_complex_s *c;
+	t_object  x_obj;
+	t_float ol_pd;
+	t_float M_pd;
+	t_float tol_pd;
+	t_float do_causal_pd;
+	t_symbol window_type;
+	phaseret_rtpghi_state_s* sta_pd;
+	ltfat_complex_s *c;
 
     t_float f;
-    
-    t_sample **out;
-    
+    t_outlet *out_real;
+    t_outlet *out_imag;
+   
 } t_rtpghi_tilde;
 
-t_int *rtpghi_tilde_perform(t_int *w)
-{
-    t_rtpghi_tilde *x = (t_rtpghi_tilde *)(w[1]);
-    const float *s =      (const float*)(w[2]);
-    int            n =             (int)(w[3]);
-    
-    ltfat_complex_s *c = (ltfat_complex_s *) x->c;
-    
-    int e = 0;
-    
-    if ((s==NULL) || ((x->sta_pd)==NULL)) {
-        post("arrays not initialised");
-    }
-    else
-    {
-        e = phaseret_rtpghi_execute_s(x->sta_pd, s, c);
-        t_sample *out=(t_sample *) x->out[0];
-        t_sample *out1= (t_sample *) x->out[1];
-        //post("e: %d", e);
-        //post("complex signal: %f + i%f\n",creal(*(c+20)), cimag(*(c+20)));
-        while (n--) {
-            
-            *out = creal(*c);
-            *out1 = cimag(*c++);
-        }
-    }
+t_int *rtpghi_tilde_perform(t_int *w) {
+	
+	t_rtpghi_tilde *x = (t_rtpghi_tilde *)(w[1]);
+	t_sample *s =      (t_sample *) (w[2]);
+	int            n =             (int)(w[5]);
 
-  return (w+4);
-}
+	ltfat_complex_s *c = (ltfat_complex_s *) x->c;
 
-void rtpghi_tilde_dsp(t_rtpghi_tilde *x, t_signal **sp)
-{
-    ltfat_int M = x->M_pd;
-    
-    //ltfat_complex_s *c = x->c;
-    post("length of c[]: %d", (M) * sizeof *(x->c));
-    post("s_n: %d", sp[0]->s_n);
-    
-    t_sample **dummy=x->out;
-    *dummy++=sp[1]->s_vec;
-    *dummy=sp[2]->s_vec;
-    
-    dsp_add(rtpghi_tilde_perform, 3,x, sp[0]->s_vec, sp[0]->s_n);
-}
+	int e = 0;
 
-void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
-{
+	if ((s==NULL) || ((x->sta_pd)==NULL)) {
+	   post("arrays not initialised");
+	}
+	else
+	{
+		e = phaseret_rtpghi_execute_s(x->sta_pd, s, c);
+		
+		t_sample *out=(t_sample *) (w[3]);
+		t_sample *out1= (t_sample *) (w[4]);
 
-  t_rtpghi_tilde *x = (t_rtpghi_tilde *)pd_new(rtpghi_tilde_class);
+		if (e == 0) {
+			while (n--) {
+				
+				if (n >= (x->M_pd)/2-1){
```

Comment:
i'm not sure i see the point of this.
does `phaseret_rtpghi_execute_s` return the full spectrum (including the (anti)symmetric 2nd half)? if so, copy all the bins (`[rifft~]` will discard the 2nd half anyhow)

Reply:
Okay, now it seems to me that both, s[] and c[] should have length of M/2 +1, because they are using always the ltfat_int M2 = M/2 +1 (in rtpghi_execute, and inside the last function rtpghimagphase(), which is combining magnitude and phase to complex array). Also in the rtpghi_offline it is visible, or am I wrong with this? [https://github.com/ltfat/phaseret/blob/master/libltfat/modules/libphaseret/src/rtpghi.c]